### PR TITLE
Fw lite mva 10 3 1

### PR DESCRIPTION
**Pull request by cbernet**
Hi Jonas! 

To be able to apply the mva electron identification in python + FWLite, ones needs: 
- to expose the electron category in python
- an infrastructure to apply the correct cut to the mva output depending on the category. 

I've implemented this for the non iso v2 only for now. Could you please have a look and test? 
If you agree with that, I can implement this feature for the other electron ids. 
I also added some documentation.

Thank you! 
Colin

### CODE REVIEW

File: RecoEgamma/ElectronIdentification/python/FWLite.py [hunk 1]
```diff
@@ -29,26 +52,62 @@ def __call__(self, ele, convs, beam_spot, rho, debug=False):
...
+                categ_id = int(name.lstrip('cutCategory'))
+                cut = cut.replace('pt','x')
+                formula = ROOT.TFormula('_'.join([self.name, wpname, name]), cut)
+                new_definitions[wpname][categ_id] = formula
+        return new_definitions
+
+    def passed(self, ele, mva, category, wp):
+        '''return true if ele passes wp'''
+        threshold = self.working_points[wp][category].Eval(ele.pt())
+        return mva > threshold  # or should it be >= ?
```
Comment (cbernet) [inline RecoEgamma/ElectronIdentification/python/FWLite.py]: Should we use >= or >  ?

Comment (guitargeek) [reply RecoEgamma/ElectronIdentification/python/FWLite.py]: It's good like this! `>` is also used in the full framework:

https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/ElectronIdentification/plugins/cuts/GsfEleMVACut.cc#L108

**Comment by guitargeek** [comment]
Ah yes, it occurs to me now what the problem is. The MVAs before Fall17V2 cut on the logistic transformation of the BDT score [1] instead of the raw BDT score. I moved to cutting on the raw BDT score because is was numerically better behaving for pt-dependent working points (after the logistic transformation the values are squeezed really too close to -1 and 1).

So thank you for this PR! I will merge, fix the Spring16 problem, test and then rebase on the master PR.

[1] https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/EgammaTools/interface/MVAValueMapProducer.h#L118

**Comment by cbernet** [comment]
Hi again Jonas, we are currently doing our analysis in `CMSSW_9_4_11_cand1` it would be great if you could backport to this release.

**Comment by guitargeek** [comment]
Hi @cbernet, ok! I made the backport:

https://github.com/guitargeek/cmssw/tree/EgammaID_FWlite_949

I also posted it in the thread corresponding to the master PR so other people can also be aware of it and use it if they like.

Cheers!
Jonas

**Pull request merged**